File: LeetCodeSolutions/LeetCode_0134.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        total = 0
        minvalue = float('inf')
        ret = -1
        for idx in range(len(gas)):
            total += gas[idx] - cost[idx]
            if total < minvalue:
                ret = idx
                minvalue = total
        return (ret + 1) % len(gas) if total >= 0 else -1

File: LeetCodeSolutions/LeetCode_0134.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost){
        int i = 0, n = gas.size(), pgas = 0, pcost = 0;
        while(i < n){
            int start = i;
            int cgas = 0, ccost = 0;
            while(i < n){
                cgas += gas[i];
                ccost += cost[i];
                if (cgas >= ccost){
                    ++i;
                }
                else break;
            }
            pgas += cgas;
            pcost += ccost;
            if (i == n && pgas >= pcost){
                return start;
            }
            ++i;
        }
        return -1;
    }
};¬